src/Compression/ICompressionCodec.cpp

```diff
@@ -125,7 +124,7 @@ UInt32 ICompressionCodec::readDecompressedBlockSize(const char * source)
 {
     UInt32 decompressed_block_size = unalignedLoadLittleEndian<UInt32>(&source[5]);
     if (decompressed_block_size == 0)
-        throw Exception(ErrorCodes::CORRUPTED_DATA, "Can't decompress data: header is corrupt with decompressed block size 0");
+        throw Exception(ErrorCodes::CANNOT_DECOMPRESS, "Can't decompress data: header is corrupt with decompressed block size 0");
```

Comment:
If this happens during unpacking stored parts of a table (e.g. during SELECT) it's still a very bad situation.

Reply:
Maybe we could pass a flag from `HTTPHandler` to `CompressedReadBuffer` and further, so codecs will throw `CANNOT_DECOMPRESS` for user's input and `CORRUPTED_DATA` for our internal data